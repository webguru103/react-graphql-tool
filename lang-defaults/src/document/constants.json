[
  {
    "id": "error-deleting-document",
    "defaultMessage": "Sorry, error deleting document!"
  },
  {
    "id": "table-head-name",
    "defaultMessage": "Name"
  },
  {
    "id": "table-head-owner",
    "defaultMessage": "Owner"
  },
  {
    "id": "table-head-last-modified",
    "defaultMessage": "Last Modified"
  },
  {
    "id": "menu-copy",
    "defaultMessage": "Copy"
  },
  {
    "id": "menu-delete",
    "defaultMessage": "Delete"
  },
  {
    "id": "no-form-selected",
    "defaultMessage": "Select at least one form"
  }
]