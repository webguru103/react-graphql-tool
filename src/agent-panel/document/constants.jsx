import React from 'react';
import { FormattedMessage } from 'react-intl';

// TODO: create utility function to create FormattedMessages on demand, so all these are not stored in memory
// eg. to call translated text, just call function with id: trans('id', 'd'efaultMessage'), and will return:
// <FormattedMessage id="id" defaultMessage="defaultMessage" />

const DEFAULT_PAGE_SIZE = 15;

export default {
  DEFAULT_PAGE_SIZE,
  ERROR_DELETING_DOCUMENT: <FormattedMessage id="error-deleting-document" defaultMessage="Sorry, error deleting document!" />,
  TABLE_HEAD_NAME: <FormattedMessage id="table-head-name" defaultMessage="Name" />,
  TABLE_HEAD_OWNER: <FormattedMessage id="table-head-owner" defaultMessage="Owner" />,
  TABLE_HEAD_LAST_MODIFIED: <FormattedMessage id="table-head-last-modified" defaultMessage="Last Modified" />,
  MENU_COPY: <FormattedMessage id="menu-copy" defaultMessage="Copy" />,
  MENU_DELETE: <FormattedMessage id="menu-delete" defaultMessage="Delete" />,
  NO_FORM_SELECTED: <FormattedMessage id="no-form-selected" defaultMessage="Select at least one form" />,
};
