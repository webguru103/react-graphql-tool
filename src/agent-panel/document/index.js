export { default as Documents } from './document';
