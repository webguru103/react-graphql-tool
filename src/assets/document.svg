<svg width="13" height="16" viewBox="0 0 13 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M7.50151 0.0559845C7.60881 0.101017 7.70293 0.171212 7.7765 0.259186L12.7407 5.2234C12.8288 5.29706 12.8991 5.39134 12.9442 5.49883C12.9837 5.58858 13.001 5.68 13 5.76927L13 5.77778V13C13 14.5955 11.7066 15.8889 10.1111 15.8889H2.88889C1.2934 15.8889 0 14.5955 0 13V2.88889C0 1.2934 1.2934 0 2.88889 0H7.22222L7.23073 4.91183e-05C7.32012 -0.00105195 7.41165 0.0163218 7.50151 0.0559845ZM7.94444 2.46175V4.33333C7.94444 4.73221 8.26779 5.05556 8.66667 5.05556H10.5382L7.94444 2.46175ZM1.44444 2.88889C1.44444 2.09114 2.09114 1.44444 2.88889 1.44444H6.5V4.33333C6.5 5.52995 7.47005 6.5 8.66667 6.5H11.5556V13C11.5556 13.7977 10.9089 14.4444 10.1111 14.4444H2.88889C2.09114 14.4444 1.44444 13.7977 1.44444 13V2.88889Z" fill="#A2A9B4"/>
</svg>
