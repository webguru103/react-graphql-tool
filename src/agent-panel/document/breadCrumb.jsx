import React from 'react';
import { withRouter } from 'react-router-dom';
import { withTransactionName } from './api/document.service';
import { compose } from '../../utility';

const TransactionBreadcrumbC = ({ transactionName }: {transactionName: string}) => (
  <span>{transactionName}</span>
);

export default compose(
  withRouter,
  withTransactionName,
)(TransactionBreadcrumbC);
