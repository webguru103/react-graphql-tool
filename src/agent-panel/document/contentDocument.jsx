import IconButton from '@material-ui/core/IconButton';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import DocumentIcon from '@material-ui/icons/Description';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
import moment from 'moment';
import * as React from 'react';
import { withRouter } from 'react-router-dom';
import type { Match } from 'react-router';
import { withCopyDocument } from './api/document.service';
import c from './constants';
import styles from './contentDocument.styles';
import DeleteDocumentDialog from './deleteDocumentDialog';
import type { DocumentType } from './flowTypes';
import { compose } from '../../utility';
import { withDialog } from '../../shared/dialog/withDialog';

type Props = {
  documents: Array<DocumentType>,
  classes: Object,
  copyDocument: Function,
  onDocumentClick: Function,
  createDialog: Function,
  match: Match,
}

export class ContentDocumentC extends React.PureComponent<Props, *> {

  state = {
    anchorEl: null,
    anchorId: null,
    activeDocument: {},
  };

  handleMenuClick = (e: Event, id: string) => {
    e.stopPropagation();

    this.setState({
      anchorEl: e.currentTarget,
      anchorId: id,
      activeDocument: (this.props.documents.filter(document => document.id === id))[0],
    });
  };

  handleDelete = (e: Event) => {
    const { createDialog, match } = this.props;
    const { activeDocument } = this.state;
    this.handleClose(e);
    createDialog({
      dialogContent: <DeleteDocumentDialog document={activeDocument} transactionId={match.params.transactionId} />,
    });
  };

  handleClose = (e: Event) => {
    e.stopPropagation();
    this.setState({ anchorEl: null });
  };

  handleCopy = (e: Event) => {
    const { match } = this.props;
    this.props.copyDocument({
      id: this.state.anchorId,
      transactionId: match.params.transactionId,
    });
    this.handleClose(e);
  };

  render() {
    const {
      documents,
      classes,
      onDocumentClick,
    } = this.props;
    const { anchorEl } = this.state;
    return (
      <React.Fragment>
        <Paper classes={{ root: classes.contentHeader }} elevation={0}>
          <span>{c.TABLE_HEAD_NAME}</span>
          <span>{c.TABLE_HEAD_OWNER}</span>
          <span>{c.TABLE_HEAD_LAST_MODIFIED}</span>
        </Paper>
        {documents && documents.map(d => (
          <Paper
            key={d.id}
            onClick={() => onDocumentClick(d.id)}
            classes={{ root: classes.document }}
            elevation={1}
          >
            <div className={classes.contentCell}>
              <DocumentIcon classes={{ root: classes.contentDocumentIcon }} />
              <span>{d.name}</span>
            </div>
            <div className={classes.contentCell}>
              <span>-</span>
            </div>
            <div className={`${classes.contentCell} ${classes.contentCellWithButton}`}>
              {d.updatedAt ? moment(d.updatedAt).format('MMMM Do YYYY, h:mm A') : '-'}
              <IconButton
                onClick={e => this.handleMenuClick(e, d.id)}
              >
                <MoreHorizIcon />
              </IconButton>
            </div>
          </Paper>
        ))}
        <Menu
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={this.handleClose}
        >
          <MenuItem onClick={this.handleCopy}>{c.MENU_COPY}</MenuItem>
          <MenuItem onClick={this.handleDelete}>{c.MENU_DELETE}</MenuItem>
        </Menu>
      </React.Fragment>
    );
  }

}

export default compose(
  withCopyDocument,
  withStyles(styles),
  withDialog,
  withRouter,
)(ContentDocumentC);
