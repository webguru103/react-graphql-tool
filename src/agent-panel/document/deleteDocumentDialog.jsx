import React from 'react';
import DialogButtonBottom from '../../shared/dialogButtonBottom/dialogButtonBottom';
import { withDeleteDocument } from './api/document.service';
import { compose, pro } from '../../utility';
import c from './constants';
import type { DocumentType } from './flowTypes';
import { withDialog } from '../../shared/dialog/withDialog';

type Props = {
  document: DocumentType,
  deleteDocument: Function,
  closeDialog: Function,
  transactionId: String,
}

export class DeleteDocumentDialogC extends React.Component<Props, *> {

  state = {
    networkError: '',
  };

  delete = async () => {
    const {
      document,
      deleteDocument,
      closeDialog,
      transactionId,
    } = this.props;

    const [err] = await pro(deleteDocument({ id: document.id, transactionId }));

    if (err) {
      this.setState({
        networkError: c.ERROR_DELETING_DOCUMENT,
      });
    } else {
      closeDialog();
    }
  };

  render() {
    return (
      <DialogButtonBottom
        title="Delete Item"
        content={(
          <div>
            <div>Are you sure you want to delete this document?</div>
            {this.state.networkError && <div style={{ marginTop: '10px', color: '#f44336' }}>{this.state.networkError}</div>}
          </div>
)}
        actionButtonText="Continue"
        onSubmit={this.delete}
      />
    );
  }

}

export default compose(
  withDeleteDocument,
  withDialog,
)(DeleteDocumentDialogC);
