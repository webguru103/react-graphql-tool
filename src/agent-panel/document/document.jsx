import React from 'react';
import { withRouter } from 'react-router-dom';
import type { Match, RouterHistory } from 'react-router';
import CircularProgress from '@material-ui/core/CircularProgress';
import { withStyles } from '@material-ui/core/styles';
import Content from './contentDocument';
import styles from './document.styles';
import { DEBOUNCE_TIMEOUT } from '../../configurations';
import type { DocumentType } from './flowTypes';
import { withDocuments } from './api/document.service';
import { compose } from '../../utility';

type Props = {
  classes: Object,
  fetchMore: Function,
  documents: Array<DocumentType>,
  history: RouterHistory,
  match: Match,
  onFilter: Function,
  filterInput: string,
};

export class DocumentC extends React.Component<Props, *> {

  static getDerivedStateFromProps(nextProps: Props, prevState: any) {
    if (nextProps.documents !== prevState.documents) {
      return ({ documents: nextProps.documents });
    }
    return null;
  }

  state = {
    loadingMore: false,
    filtering: false,
    documents: this.props.documents,
    page: 0,
  };

  containerRef = null;

  componentDidMount() {
    if (this.containerRef) {
      window.addEventListener('scroll', this.fetchMoreOnScroll);
    }
  }

  componentDidUpdate(prevProps: Props) {
    if (prevProps.filterInput !== this.props.filterInput) {
      this.handleDocumentsFilter();
    }
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.fetchMoreOnScroll);
  }

  onDocumentClick = (id: string) => {
    const { history, match } = this.props;
    history.push(`${match.url}/${id}`);
  };

  fetchMoreOnScroll = () => {
    const { match } = this.props;

    if (
      document.documentElement && document.body
      && (document.body.offsetHeight + document.documentElement.scrollTop) >= document.documentElement.scrollHeight) {
      if (!this.state.loadingMore) {
        this.setState({ loadingMore: true });
        setTimeout(async () => {
          await this.props.fetchMore({
            id: match.params.transactionId,
            page: this.state.page + 1,
            filter: this.props.filterInput,
          });
          this.setState(prevState => ({ loadingMore: false, page: prevState.page + 1 }));
        }, Number(DEBOUNCE_TIMEOUT));
      }
    }
  };

  handleDocumentsFilter = () => {
    this.setState({ filtering: true });
    setTimeout(async () => {
      await this.props.onFilter({ filter: this.props.filterInput });
      this.setState({ filtering: false });
    }, Number(DEBOUNCE_TIMEOUT));
  };

  render() {
    const {
      documents, loadingMore, filtering,
    } = this.state;
    const { classes, filterInput } = this.props;
    return (
      <div ref={r => (this.containerRef = r)} className={classes.document}>
        <Content documents={documents} onDocumentClick={this.onDocumentClick} filtering={filtering} filterInput={filterInput} />
        {filtering
          && (
          <div className={classes.filteringProgress}>
            <CircularProgress />
          </div>
          )
        }
        <div className={classes.loadMore}>
          {loadingMore && <CircularProgress />}
        </div>
      </div>
    );
  }

}

export default compose(
  withStyles(styles, { withTheme: true }),
  withRouter,
  withDocuments,
)(DocumentC);
