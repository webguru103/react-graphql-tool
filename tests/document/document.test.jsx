import React from 'react';
import { DocumentC } from '../../src/agent-panel/document/document';
import { shallowWrap, mountWrap } from '../test-helpers/contextWrapper';
import documentMockData from '../test-helpers/documentMockData';

describe('DocumentC', () => {
  let props;
  let component;
  const wrappedShallow = (lang = 'en') => {
    component = shallowWrap(<DocumentC {...props} />, lang);
    return component;
  };

  const wrappedMount = (lang = 'en') => {
    component = mountWrap(<DocumentC {...props} />, lang);
    return component;
  };

  beforeEach(() => {
    props = {
      classes: {},
      documents: documentMockData,
    };
    if (component) component.unmount();
  });

  test('should render', () => {
    const wrapper = wrappedShallow();
    expect(wrapper).toMatchSnapshot();
  });

  test('should find row name on page for each row passed in', () => {
    const wrapper = wrappedMount();
    documentMockData.forEach((item) => {
      expect(wrapper.contains(item.name)).toBe(true);
    });
  });
});
