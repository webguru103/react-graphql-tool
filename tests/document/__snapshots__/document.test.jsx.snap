// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentC should render 1`] = `
<div>
  <Apollo(CopyDocumentHOC)
    documents={
      Array [
        Object {
          "createdAt": "01/01/2018",
          "id": "id1",
          "name": "Document 1",
          "updatedAt": "01/01/2018",
        },
        Object {
          "createdAt": "01/01/2018",
          "id": "id2",
          "name": "Document 2",
          "updatedAt": "01/01/2018",
        },
        Object {
          "createdAt": "01/01/2015",
          "id": "id3",
          "name": "Document 3",
          "updatedAt": "01/01/2018",
        },
        Object {
          "createdAt": "01/01/2015",
          "id": "id4",
          "name": "Document 4",
          "updatedAt": "01/01/2018",
        },
      ]
    }
    filtering={false}
    onDocumentClick={[Function]}
  />
  <div />
</div>
`;
